src/chess/position.cc

```diff
@@ -128,4 +158,39 @@ uint64_t PositionHistory::HashLast(int positions) const {
   return HashCat(hash, Last().GetRule50Ply());
 }
 
+// PrintFen outputs a FEN notation of the board.
+// based on
+// https://en.wikipedia.org/wiki/Forsyth%E2%80%93Edwards_Notation#Examples
+std::string Position::GetFen() const {
```

Comment:
This function doesn't really access (or has to access) any private member variables, consider making it a free-standing function rather than a member. <https://www.drdobbs.com/cpp/how-non-member-functions-improve-encapsu/184401197>, <https://flylib.com/books/en/2.887.1.37/1/>